cmd/juju/cloud/addcredential.go

```diff
@@ -339,26 +356,31 @@ func (c *addCredentialCommand) interactiveAddCredential(ctxt *cmd.Context, schem
 	return nil
 }
 
-func (c *addCredentialCommand) finalizeProvider(ctxt *cmd.Context, authType jujucloud.AuthType, attrs map[string]string) (*jujucloud.Credential, error) {
-	cloudEndpoint := c.cloud.Endpoint
-	cloudStorageEndpoint := c.cloud.StorageEndpoint
-	cloudIdentityEndpoint := c.cloud.IdentityEndpoint
-	if len(c.cloud.Regions) > 0 {
-		// NOTE(axw) we use the first region in the cloud,
-		// because this is all we need for Azure right now.
-		// Each region has the same endpoints, so it does
-		// not matter which one we use. If we expand
-		// credential generation to other providers, and
-		// they do have region-specific endpoints, then we
-		// should prompt the user for the region to use.
-		// That would be better left to the provider, though.
-		region := c.cloud.Regions[0]
+func FinalizeProvider(ctxt *cmd.Context, cloud *jujucloud.Cloud, regionName, defaultRegion string, authType jujucloud.AuthType, attrs map[string]string) (*jujucloud.Credential, error) {
```

Comment:
Would be nice to have an explicit test for this or remain unexported.

Reply:
was there previously but sure, I've unexported.